webapp/certified/views.py

```diff
@@ -0,0 +1,352 @@
+                    for device in devices:
+                        release_details["components"][device_category].append(
+                            {
+                                "name": (
+                                    f"{device['make']} {device['name']}"
+                                    f" {device['subproduct_name']}"
+                                ),
+                                "bus": device["bus"],
+                                "identifier": device["identifier"],
+                            }
+                        )
+
+    # default to category, which contains the least specific form_factor
+    form_factor = model_release and model_release.get(
+        "form_factor", model_release["category"]
+    )
+
+    return render_template(
+        "certified/model-details.html",
+        canonical_id=canonical_id,
+        name=model_release["model"],
+        category=model_release["category"],
+        form_factor=form_factor,
+        vendor=model_release["make"],
+        major_release=model_release["certified_release"],
+        release_details=release_details,
+        has_enabled_releases=has_enabled_releases,
+        components=component_summaries,
+    )
+
+
+def certified_home():
+
+    certified_releases = api.certified_releases(limit="0")["objects"]
+    certified_makes = api.certified_makes(limit="0")["objects"]
+
+    # Laptop section
+    laptop_releases = []
+    laptop_vendors = []
+
+    # Desktop section
+    desktop_releases = []
+    desktop_vendors = []
+
+    # SoC section
+    soc_releases = []
+    soc_vendors = []
+
+    # IoT section
+    iot_releases = []
+    iot_vendors = []
+
+    # Search results filters
+    all_releases = []
+    all_vendors = []
+
+    for release in certified_releases:
+        version = release["release"]
+
+        if release not in all_releases:
+            all_releases.append(version)
+            all_releases = sorted(all_releases, reverse=True)
+
+        if int(release["laptops"]) > 0:
+            laptop_releases.append(release)
+
+        if int(release["desktops"]) > 0:
+            desktop_releases.append(release)
+
+        if int(release["smart_core"] > 1):
+            iot_releases.append(release)
+
+        if int(release["soc"] > 1):
+            soc_releases.append(release)
+
+    for vendor in certified_makes:
+        make = vendor["make"]
+
+        if make not in all_vendors:
+            all_vendors.append(make)
+            all_vendors = sorted(all_vendors)
+
+        if int(vendor["laptops"]) > 0:
+            laptop_vendors.append(vendor)
+
+        if int(vendor["desktops"]) > 0:
+            desktop_vendors.append(vendor)
+
+        if int(vendor["smart_core"] > 1):
+            iot_vendors.append(vendor)
+
+        if int(vendor["soc"] > 1):
+            soc_vendors.append(vendor)
+
+    # Server section
+    server_releases = {}
+    server_vendors = api.vendor_summaries_server()["vendors"]
```

Comment:
why not treat servers the same way the other categories are treated? if you do that you can drop the expensive api call.

Reply:
We discussed this and decided to leave it the same for now since the results are counted differently on the old site and we want to reproduce what the old site does.

Reply:
As discussed, data is no the same.